README.md

```diff
@@ -134,18 +140,89 @@ $ tree /tmp/mounted_image
 2 directories, 2 files
 ```
 
-If you want to enable `fs-verity` checks, run
+For additional mount options, run `cargo run -- mount -h`.
+
+### Mounting with fs-verity enabled
+If you want to mount the filesystem with `fs-verity` authenticity protection, first enable `fs-verity` by running:
+```
+$ cargo run --release -- enable-fs-verity /tmp/puzzlefs-image puzzlefs_example 9ac9abc098870c55cc61431dae8635806273d8f61274d34bec062560e79dc2f5
+```
+This makes the data and metadata files readonly. Any reads of corrupted data will fail.
+
+Then run mount with the `--digest` option:
 ```
 $ cargo run --release -- mount --digest 9ac9abc098870c55cc61431dae8635806273d8f61274d34bec062560e79dc2f5 /tmp/puzzlefs-image puzzlefs_example /tmp/mounted_image
 ```
+PuzzleFS now ensures that each file it opens has fs-verity enabled and that the
+fs-verity measurement matches the fs-verity data stored in the manifest. The
+image manifest's fs-verity digest is compared with the digest passed on the
+command line via the `--digest` option.
+
+This only works if `fsverity` is [supported and
+enabled](https://www.kernel.org/doc/html/latest/filesystems/fsverity.html#filesystem-support)
+in the underlying filesystem on which the puzzlefs image resides.  Otherwise
+you might get an error like this when running `enable-fs-verity`:
+```
+Error: fs error: Inappropriate ioctl for device (os error 25)
+
+Caused by:
+    Inappropriate ioctl for device (os error 25)
+```
+
+To check wheter fs-verity is enabled, use `tune2fs`:
+```
+$ mount | grep -w '/'
+/dev/mapper/MyVolGroup-root on / type ext4 (rw,relatime)
+
+$ sudo tune2fs -l /dev/mapper/MyVolGroup-root | grep verity
+Filesystem features:      has_journal ext_attr resize_inode dir_index filetype needs_recovery extent 64bit flex_bg sparse_super large_file huge_file dir_nlink extra_isize metadata_csum verity
+```
 
-This only works if `fsverity` is [supported and enabled](https://www.kernel.org/doc/html/latest/filesystems/fsverity.html#filesystem-support) in the underlying filesystem on which the puzzlefs image resides.
-Otherwise you might get an error like this:
+To set up an 1MB loop device with an ext4 filesystem which supports `fs-verity`
+and mount it under `/mnt`, run:
+```
+$ mktemp -u
+/tmp/tmp.2CDDHVPLXp
+
+$ touch /tmp/tmp.2CDDHVPLXp
+
+$ dd if=/dev/zero of=/tmp/tmp.2CDDHVPLXp bs=1k count=1024
+1024+0 records in
+1024+0 records out
+1048576 bytes (1.0 MB, 1.0 MiB) copied, 0.00203188 s, 516 MB/s
+
+$ sudo losetup -f --show /tmp/tmp.2CDDHVPLXp
+/dev/loop1
+
+$ sudo mkfs -t ext4 -F -b4096 -O verity /dev/loop1
+mke2fs 1.47.0 (5-Feb-2023)
+
+Filesystem too small for a journal
+Discarding device blocks: done
+Creating filesystem with 256 4k blocks and 128 inodes
+
+Allocating group tables: done
+Writing inode tables: done
+Writing superblocks and filesystem accounting information: done
+
+$ sudo mount /dev/loop1 /mnt
+
+$ sudo chown -R $(id -u):$(id -g) /mnt
+
+$ sudo tune2fs -l /dev/loop1 | rg verity
```

Comment:
what is rg?

Reply:
https://github.com/BurntSushi/ripgrep but I should have used grep here as I do in all the other cases